crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -1857,11 +1868,17 @@ pub struct RpcServerHandle {
     ws: Option<ServerHandle>,
     ipc_endpoint: Option<String>,
     ipc: Option<ServerHandle>,
+    jwt_secret: Option<JwtSecret>,
 }
 
 // === impl RpcServerHandle ===
 
 impl RpcServerHandle {
+    /// Configures the JWT secret for authentication.
+    pub fn with_jwt_secret(mut self, secret: Option<JwtSecret>) -> Self {
+        self.jwt_secret = secret;
+        self
+    }
```

Comment:
we don't want to support setting this externally, so this can be removed

but we need to install the auth header in the client functions if the jwt is present, see `http/ws_client`

see also:

https://github.com/paradigmxyz/reth/blob/3ab1afc9aa11cfc50236509a217b26d522fbb9b2/crates/rpc/rpc-builder/src/auth.rs#L306-L318

https://github.com/paradigmxyz/reth/blob/3ab1afc9aa11cfc50236509a217b26d522fbb9b2/crates/rpc/rpc-builder/src/auth.rs#L320-L326